aws-datastore/src/main/java/com/amplifyframework/datastore/syncengine/LastSyncMetadata.java

```diff
@@ -39,40 +41,49 @@ public final class LastSyncMetadata implements Model {
     private final @ModelField(targetType = "String", isRequired = true) String modelClassName;
     private final @ModelField(targetType = "AWSTimestamp", isRequired = true) Long lastSyncTime;
     private final @ModelField(targetType = "String", isRequired = true) String lastSyncType;
+    private final @ModelField(targetType = "String") QueryPredicate syncExpression;
 
     @SuppressWarnings("checkstyle:ParameterName") // The field is named "id" in the model; keep it consistent
-    private LastSyncMetadata(String id, String modelClassName, Long lastSyncTime, SyncType syncType) {
+    private LastSyncMetadata(String id, String modelClassName, Long lastSyncTime,
+                             SyncType syncType, @Nullable QueryPredicate syncExpression) {
         this.id = id;
         this.modelClassName = modelClassName;
         this.lastSyncTime = lastSyncTime;
         this.lastSyncType = syncType.name();
+        this.syncExpression = syncExpression;
     }
 
     /**
      * Creates an instance of an {@link LastSyncMetadata}, indicating that the provided
      * model has been base sync'd, and that the last sync occurred at the given time.
      * @param modelClassName Name of model
      * @param lastSyncTime Last time it was synced
+     * @param syncExpression the corresponding sync expression being used during last sync
      * @param <T> t type of Model.
      * @return {@link LastSyncMetadata} for the model class
      */
+    @InternalApiWarning
     public static <T extends Model> LastSyncMetadata baseSyncedAt(@NonNull String modelClassName,
```

Comment:
This is a breaking change. Need to leave alone and created method overload. Internally, we should check for usages and make sure we are't using this method. Can mark this one as deprecated.

Reply:
Make sense!